### PR TITLE
Docs: Exclude decorators by default from source

### PR DESCRIPTION
**Pull request by tmeasday**
## What I did

Due to reordering in #21182 the source decorator comes *before* any user decorator in `preview.js`. This means that we commonly trigger the bug in #21649.

Really it was a bug that was fixed by #21182 that the user's decorators *weren't* visible in the source by default given we weren't excluding them.

So it makes more sense to default this to true and people can opt out in cases where they want to show the decorator.

## How to test

Add the following to a React sandbox's `preview.tsx`:

```tsx
  decorators: [
    (S) => <div><S/></div>
  ]
```

Then check the source block of a story:

image

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -75,7 +75,8 @@
     - [Dropped addon-docs manual babel configuration](#dropped-addon-docs-manual-babel-configuration)
     - [Dropped addon-docs manual configuration](#dropped-addon-docs-manual-configuration)
     - [Autoplay in docs](#autoplay-in-docs)
-    - [Removed STORYBOOK_REACT_CLASSES global](#removed-storybook_react_classes-global)
+    - [Removed STORYBOOK\_REACT\_CLASSES global](#removed-storybook_react_classes-global)
```
Comment (shilman) [inline MIGRATION.md:78]: Bad formatting

**Review by shilman — DISMISSED** [review]
Aside from some markdown issues LGTM. Thanks @tmeasday !

File: MIGRATION.md [hunk 2]
```diff
@@ -31,7 +31,7 @@
     - [Addon-a11y: Removed deprecated withA11y decorator](#addon-a11y-removed-deprecated-witha11y-decorator)
   - [7.0 Vite changes](#70-vite-changes)
     - [Vite builder uses Vite config automatically](#vite-builder-uses-vite-config-automatically)
-    - [Vite cache moved to node_modules/.cache/.vite-storybook](#vite-cache-moved-to-node_modulescachevite-storybook)
+    - [Vite cache moved to node\_modules/.cache/.vite-storybook](#vite-cache-moved-to-node_modulescachevite-storybook)
```
Comment (tmeasday) [inline MIGRATION.md:34]:
```suggestion
    - [Vite cache moved to node_modules/.cache/.vite-storybook](#vite-cache-moved-to-node_modulescachevite-storybook)
```

File: MIGRATION.md (hunk 1 above)
Comment (tmeasday) [reply MIGRATION.md:78]:
```suggestion
    - [Removed STORYBOOK_REACT_CLASSES global](#removed-storybook_react_classes-global)
```

**Review by valentinpalkovic — APPROVED** [review]

**Review by shilman — APPROVED** [review]

**Pull request merged**